velox/functions/sparksql/benchmarks/CompareBenchmark.cpp

```diff
@@ -0,0 +1,84 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+
+#include "velox/benchmarks/ExpressionBenchmarkBuilder.h"
+#include "velox/functions/sparksql/Register.h"
+
+using namespace facebook;
+
+using namespace facebook::velox;
+
+int main(int argc, char** argv) {
+  folly::Init init(&argc, &argv);
+  functions::sparksql::registerFunctions("");
+
+  ExpressionBenchmarkBuilder benchmarkBuilder;
+  const vector_size_t vectorSize = 1000;
+  auto vectorMaker = benchmarkBuilder.vectorMaker();
+  auto shortDecimalInput = vectorMaker.flatVector<int64_t>(
+      vectorSize,
+      [&](auto j) { return 123456789 * j; },
+      nullptr,
+      DECIMAL(18, 6));
+  auto longDecimalInput = vectorMaker.flatVector<int128_t>(
+      vectorSize,
+      [&](auto j) {
+        return facebook::velox::HugeInt::build(12345 * j, 56789 * j + 12345);
+      },
+      nullptr,
+      DECIMAL(38, 16));
+  auto timestampInput =
+      vectorMaker.flatVector<Timestamp>(vectorSize, [&](auto j) {
+        return Timestamp(1695859694 + j / 1000, j % 1000 * 1'000'000);
+      });
+
+  benchmarkBuilder
+      .addBenchmarkSet(
+          "compare",
+          vectorMaker.rowVector(
+              {"c0", "c1"}, {shortDecimalInput, longDecimalInput}))
+      .addExpression("greaterthan", "decimal_greaterthan(c0, c1)")
+      .addExpression(
+          "greaterthan_with_cast",
+          "greaterthan(cast (c0 as decimal(38, 16)), c1)")
+      .addExpression("greaterthanorequal", "decimal_greaterthanorequal(c0, c1)")
+      .addExpression(
+          "greaterthanorequal_with_cast",
+          "greaterthanorequal(cast (c0 as decimal(38, 16)), c1)")
+      .addExpression("lessthan", "decimal_lessthan(c0, c1)")
+      .addExpression(
+          "lessthan_with_cast", "lessthan(cast (c0 as decimal(38, 16)), c1)")
+      .addExpression("lessthanorequal", "decimal_lessthanorequal(c0, c1)")
+      .addExpression(
+          "lessthanorequal_with_cast",
+          "lessthanorequal(cast (c0 as decimal(38, 16)), c1)")
+      .addExpression("equalto", "decimal_equalto(c0, c1)")
+      .addExpression(
+          "equalto_with_cast", "equalto(cast (c0 as decimal(38, 16)), c1)")
+      .addExpression("notequalto", "decimal_notequalto(c0, c1)")
+      .addExpression(
+          "notequalto_with_cast",
+          "not(equalto(cast (c0 as decimal(38, 16)), c1))")
+      .withIterations(100)
+      .disableTesting();
```

Comment:
Why do we disable testing? Don't we want to make sure that expression with and without cast return same results?

Reply:
I see. This is added by mistake. Removed.